spirv_msl.cpp

```diff
@@ -398,6 +510,19 @@ void CompilerMSL::emit_header()
 	statement("");
 	statement("using namespace metal;");
 	statement("");
+    emit_msl_defines();
+}
+
+void CompilerMSL::emit_msl_defines()
+{
+    statement("// Standard GLSL->MSL redefinitions");
+    statement("#define discard discard_fragment()");
+    statement("#define dFdy dfdy");
+    statement("#define dFdx dfdy");
+    statement("#define atan(x,y) atan2((y),(x))");
+    statement("inline bool greaterThan(float2 a,float2 b) { return all(a>b) ? true : false; }");
```

Comment:
Agreed. Being added for specific customer requirements right now. A better approach to mapping. GLSL->MSL functions needs to be developed.